MonoDevelop.MSBuild.Editor/Completion/MSBuildCompletionSource.cs

```diff
@@ -107,12 +123,132 @@ async Task<MSBuildCompletionSessionContext> GetSessionContext (IAsyncCompletionS
 			return new CompletionContext (ImmutableArray<CompletionItem>.Empty.AddRange (items));
 		}
 
+		//Dictionary<int, string> cache = new Dictionary<int, string> ();
+
+		protected override async Task<CompletionContext> GetAttributeValueCompletionsAsync (
+			IAsyncCompletionSession session,
+			SnapshotPoint triggerLocation,
+			List<XObject> nodePath,
+			IAttributedXObject attributedObject,
+			XAttribute attribute,
+			CancellationToken token)
+		{
+			/*if (!attribute.NameEquals ("include", true)) {
+				return CompletionContext.Empty;
+			}*/
+
+			var context = await GetSessionContext (session, triggerLocation, token);
+			var rr = context.rr;
+			var doc = context.doc;
+
+			if (rr == null) {
+				return CompletionContext.Empty;
+			}
+
+			if (attributedObject is XElement xElement &&
+				xElement.NameEquals ("PackageReference", true)) {
+
+				switch (Regex.Replace (attribute.FriendlyPathRepresentation.ToLower (), @"[^0-9a-zA-Z]+", "")) {
+				case "include":
+					return await GetPackageNamesAsync (doc, rr);
+
+				case "version":
+					return await GetPackageVersionsAsync (doc, rr);
+				}
+			}
+
+			return CompletionContext.Empty;
+		}
+
+		private async Task<CompletionContext> GetPackageNamesAsync(MSBuildRootDocument doc,
+																MSBuildResolveResult rr)
+        {
+			if (rr == null) {
+				return CompletionContext.Empty;
+			}
+
+			var items = new List<CompletionItem> ();
+			var httpClient = new HttpClient ();
+			string result = null;
+			var searchQuery = rr.Reference.ToString ().ToLower ();
+			if (string.IsNullOrEmpty (searchQuery)) {
+				return CompletionContext.Empty;
+			}
+
+			try {
+				var cancellationPackage = new CancellationTokenSource (TimeSpan.FromSeconds (10));
+				var packageResponse = await httpClient.GetAsync ($"https://api-v2v3search-0.nuget.org/autocomplete?q={searchQuery}&skip=0&take=100", cancellationPackage.Token);
+				result = await packageResponse.Content.ReadAsStringAsync ();
+
+			} catch (Exception) {
+				return CompletionContext.Empty;
+			}
+
+			var packageJSONObject = JObject.Parse (result);
+			var packageArray = (JArray)packageJSONObject["data"];
+
+			foreach (var package in packageArray) {
+				var info = package.ToString ();
+				items.Add (CreateCompletionItem (new ItemInfo (info, "description 1", "description 2"),
+							doc, rr));
+			}
+			return new CompletionContext (ImmutableArray<CompletionItem>.Empty.AddRange (items));
+		}
+
+		private async Task<CompletionContext> GetPackageVersionsAsync (MSBuildRootDocument doc,
+																MSBuildResolveResult rr)
+		{
+			if (rr == null) {
+				return CompletionContext.Empty;
+			}
+
+			var items = new List<CompletionItem> ();
+			var httpClient = new HttpClient (); ;
+			string result = null;
+			var packageName = rr.XElement.Attributes.First.Value.ToLower ();
+
+			if (string.IsNullOrEmpty (packageName)) {
```

Comment:
Minor nitpick but you could do this before the ToLower and avoid an allocation when the string is empty.